### DISCUSSION TITLE
Reuse of Mapping

### DISCUSSION THREAD

**Discussion by umeshthakral** [mapstruct/mapstruct#3369] [Q&A]
Hi All,

I have two similar object coming from 2 different lib and both are identical inside. example:
`class ADto {
 name,
class,
roll
} `
`class BDto {
name,
class,
roll,
}`

I have to map both object to InputDTO with custom mapping
`InputDTO  {
CustomerName,
CustomerClass,
CustomerID
}`
I have defined the two mapper to do this. And Sonar is asking to remove the duplicate code. Both mapping are same except parameter 
`InputDTO  map(ADto  dto)
InputDTO  map(BDto  dto)
`
is there a way to define common mapping and use them at both methods. If it already exists then kindly let me know the example

**Comment by filiphr** [comment]
Is sonar complaining about the generated code or the defined `@Mapping` annotations?

If it is the defined annotations then you can define your own mapping meta annotation. Have a look at [Mapping Composition](https://mapstruct.org/documentation/stable/reference/html/#mapping-composition) in our documentation.
